pkg/prometheus/server/statefulset.go

```diff
@@ -201,18 +202,25 @@ func makeStatefulSet(
 			},
 		})
 	} else {
-		pvcTemplate := operator.MakeVolumeClaimTemplate(storageSpec.VolumeClaimTemplate)
-		if pvcTemplate.Name == "" {
-			pvcTemplate.Name = prompkg.VolumeName(p)
+		if reflect.DeepEqual(storageSpec.VolumeClaimTemplate, monitoringv1.EmbeddedPersistentVolumeClaim{}) {
+			storageSpec.VolumeClaimTemplates = append([]monitoringv1.EmbeddedPersistentVolumeClaim{storageSpec.VolumeClaimTemplate}, storageSpec.VolumeClaimTemplates...)
```

Comment:
As far as I could understand, you're comparing if VolumeClaimTemplate is equal com an empty object? And only if it is indeed empty, then you're appending it to array of PVClaims?

Seen in [godocs](https://pkg.go.dev/reflect#DeepEqual) of DeeplyEqual:
`Struct values are deeply equal if their corresponding fields, both exported and unexported, are deeply equal.`

```suggestion
		if !reflect.DeepEqual(storageSpec.VolumeClaimTemplate, monitoringv1.EmbeddedPersistentVolumeClaim{}) {
			storageSpec.VolumeClaimTemplates = append([]monitoringv1.EmbeddedPersistentVolumeClaim{storageSpec.VolumeClaimTemplate}, storageSpec.VolumeClaimTemplates...)
```
Looks like we're only missing the negation operator here :P